source/sync/data-access-patterns/permissions.txt

```diff
@@ -360,6 +353,222 @@ change until the next session.
      write rules. Functions would not operate on a per-document basis.
    - You cannot store permissions information (such as "which document IDs may this user access?") in the user object.
 
+
+.. _flexible-sync-roles-json-reference:
+
+Flexible Sync Roles JSON Reference
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+A **Role** object can specify a role for a specific collection or the default
+role applied to all collections. You define field-level permissions with a
+**Fields** object in the Role.
+
+.. example::
+
+   The following JSON object describes a **Role** for a team admin:
+
+   .. code-block:: json
+
+    {
+      "name": "TeamAdmin",
+      "applyWhen": { "%%user.custom_data.isAdmin": true },
+      "read": {},
+      "write": { "teamId": "%%user.custom_data.teamId" },
+      "fields": {
+        "address": {
+          "fields": {
+            "zipCode": {
+              "write": false,
+              "read": true
+            }
+          },
+          "additional_fields": {
+            "write": true
+          }
+        },
+        "name": {}
+      },
+      "additional_fields": {
+        "write": false,
+        "read": false
+      }
+    }
+  
+   The following table explains the effect of each part of the Role:
+
+   .. list-table::
+      :header-rows: 1
+      :widths: 50 50
+
+      * - Role JSON
+        - Effect
+
+      * - ``"read": {}``
+        - May access any document for reading.
+
+      * - ``"write": { "teamId": "%%user.custom_data.teamId" }``
+        - May access a document for writing if the document's ``teamId`` field
+          equals the user's custom data ``teamId``. This assumes ``teamId`` was
+          declared as a queryable field in the Sync configuration.
+
+      * - .. code-block:: json
+
+            "fields": {
+              "address": {
+                "fields": {
+                  "zipCode": {
+                    "write": false,
+                    "read": true
+                  }
+                },
+                "additional_fields": {
+                  "write": true
+                }
+              },
+              "name": {}
+            },
+
+        - May write to any field of the ``address`` embedded object except
+          ``zipCode``. May write to the ``name`` field.
+      
+      * - .. code-block:: json
+
+            "additional_fields": {
+              "write": false,
+              "read": false
+            }
+
+        - May not read or write any other fields despite having document-level
+          access to the document.
+          
+
+
+A **Role** object has the following properties:
```

Comment:
I think this whole section might benefit from this coming _before_ the specific example.

Reply:
Cool. I had gone back and forth. I think I agree with you, let's try it!